posts/2021-03-24-microprofile40-open-liberty-21003.adoc

```diff
@@ -0,0 +1,441 @@
+
+==== SimpleTimer
+The SimpleTimer metric now tracks and reports the highest and lowest recorded time duration of the previous complete minute. See link:https://download.eclipse.org/microprofile/microprofile-metrics-3.0/apidocs/index.html?org/eclipse/microprofile/metrics/SimpleTimer.html[SimpleTimer Javadoc] for more information.
+
+==== Timer
+The Time metric now tracks and reports the total elapsed time duration. See link:https://download.eclipse.org/microprofile/microprofile-metrics-3.0/apidocs/index.html?org/eclipse/microprofile/metrics/Timer.html[Timer Javadoc] for more information.
+
+==== REST metric
+A new `REST.request.unmappedException.total` metric that is backed by a counter metric has been introduced. Similar to the `REST.request` metric, there will be one unique metric for each REST endpoint identified by a class and method label. The new REST metric will count the amount of times the request ends in an unmapped exception. The REST.request metric corresponding to this REST endpoint will not record any values if an unmapped exception has occured.
+
+==== CDI Producer
+The `@Metrics` annotation will no longer support the method target (i.e it can not be annotated on a method). Additionally, it will not support usage with CDI Producers.
+
+==== Try it now 
+
+Enable Metrics 3.0 in the `server.xml`, along with any other features you're using.
+
+[source, xml]
+----
+<featureManager>
+    <feature>mpMetrics-3.0</feature>
+</featureManager>
+----
+
+More information:
+
+* link:https://github.com/eclipse/microprofile-metrics/releases/tag/3.0[MicroProfile Metrics 3.0 Release Page (Javadoc & Spec)]
+
+* link:https://download.eclipse.org/microprofile/microprofile-metrics-3.0/microprofile-metrics-spec-3.0.html#release_notes_3_0[Release Notes]
+
+[#health-3.0]
+=== New changes in MicroProfile Health 3.0
+
+MicroProfile Health 3.0 enables you to provide your own health check procedures to be invoked by Open Liberty, to verify the health of your microservice.
+
+MicroProfile Health allows services to report their health, and it publishes the overall health status to a defined endpoint. A service reports UP if it is available and reports DOWN if it is unavailable. MicroProfile Health reports an individual service status at the endpoint and indicates the overall status as UP if all the services are UP. A service orchestrator can then use the health statuses to make decisions.
+
+MicroProfile Health checks its own health by performing necessary self-checks and then reports its overall status by implementing the API provided by MicroProfile Health. A self-check can be a check on anything that the service needs, such as a dependency, a successful connection to an endpoint, a system property, a database connection, or the availability of required resources. MicroProfile offers checks for both liveness and readiness.
```

Comment:
Actually, these two paragraphs are quite long for this blog and don't talk about what is new. Could they be removed? I assume there will be a stand-alone blog for Health as well?

Reply:
Since there is no stand-alone blog for health, maybe we do need to keep them